encode/src/main/scala/org/broadinstitute/monster/etl/encode/extract/EncodeExtractions.scala

```diff
@@ -0,0 +1,93 @@
+package org.broadinstitute.monster.etl.encode.extract
+
+import io.circe.JsonObject
+import org.broadinstitute.monster.etl.encode.extract.client.EncodeClient
+import cats.implicits._
+import org.apache.beam.sdk.transforms.GroupIntoBatches
+import org.apache.beam.sdk.values.KV
+import com.spotify.scio.values.SCollection
+import scala.collection.JavaConverters._
+
+class EncodeExtractions(client: EncodeClient) {
+
+  def extractEntities(
+    entryName: String,
+    entityType: String
+  ): SCollection[List[(String, String)]] => SCollection[JsonObject] =
+    _.transform(s"extractEntities - $entryName") {
+      _.flatMap { params =>
+        //generic operation on SCollection[List[(String, String)]] for all steps of extractions
+        client
+          .search(
+            entityType,
+            List("frame" -> "object", "status" -> "released") ::: params
+          )
+          .compile
+          .toList
+          .unsafeRunSync()
+      }
+    }
+
+  def getExperimentSearchParams(
+    entryName: String
+  ): SCollection[String] => SCollection[List[(String, String)]] =
+    _.transform(s"getExperimentSearchParams - $entryName") {
+      for {
+        assayType <- _
```

Comment:
😮 had no clue this would work.

Since this `for` is short, it'd be clearer as a normal `map`:
```scala
_.map(assayType => List("assay_title" -> assay_type)
```